Sources/Networking/HTTPClient/HTTPClient.swift

```diff
@@ -518,6 +540,40 @@ private extension HTTPClient {
         return self.signing
     }
 
+    private func trackHttpRequestPerformedIfNeeded(request: Request,
+                                                   requestStartTime: Date,
+                                                   result: Result<VerifiedHTTPResponse<Data>, NetworkError>?) {
+        if #available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *) {
+            if let diagnosticsTracker = self.diagnosticsTracker {
+                let responseTime = self.dateProvider.now().timeIntervalSince(requestStartTime)
+                if let result = result {
+                    let requestPathName = request.httpRequest.path.name
+                    Task(priority: .background) {
+                        switch result {
+                        case let .success(response):
+                            let httpStatusCode = response.httpStatusCode.rawValue
+                            let verificationResult = response.verificationResult
+                            await diagnosticsTracker.trackHttpRequestPerformed(endpointName: requestPathName,
+                                                                               responseTime: responseTime,
+                                                                               wasSuccessful: true,
+                                                                               responseCode: httpStatusCode,
+                                                                               resultOrigin: response.origin,
+                                                                               verificationResult: verificationResult)
+                        // swiftlint:disable:next empty_enum_arguments
+                        case .failure(_):
+                            await diagnosticsTracker.trackHttpRequestPerformed(endpointName: requestPathName,
+                                                                               responseTime: responseTime,
+                                                                               wasSuccessful: false,
+                                                                               responseCode: -1,
+                                                                               resultOrigin: nil,
+                                                                               verificationResult: .notRequested)
```

Comment:
Again unsure on conventions here but `.notRequested` seems possibly incorrect?

Reply:
Yeah, the idea here is that upon a network error, the verification doesn't even happen, so we default to `.notRequested`. We could default to something else to differentiate, but we already have the `statusCode` property to be able to differentiate those and filter in grafana as needed. We also avoid adding new cardinality to the grafana counters. Lmk if you feel we should have something else though!